<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RoomConfigDialog</class>
 <widget class="QDialog" name="RoomConfigDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1091</width>
    <height>591</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Room Config</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="rightMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QScrollArea" name="scrollArea">
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="sizeAdjustPolicy">
        <enum>QAbstractScrollArea::AdjustToContents</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>781</width>
          <height>778</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_UI">
           <property name="leftMargin">
            <number>6</number>
           </property>
           <property name="rightMargin">
            <number>6</number>
           </property>
           <property name="bottomMargin">
            <number>20</number>
           </property>
           <item>
            <widget class="QLabel" name="tilesetLabel">
             <property name="text">
              <string>Tileset:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="ComboBox_TilesetID"/>
           </item>
           <item>
            <widget class="QCheckBox" name="CheckBox_BGLayerEnable">
             <property name="text">
              <string>Enable Background Layer</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="backgroundLabel">
             <property name="text">
              <string>Use existing Background Layer:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="ComboBox_BGLayerPicker">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="editable">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label_BGLayerScrolling">
               <property name="text">
                <string>Background Layer Scrolling Flag:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_CurBGLayerScrollingType">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_BGLayerScrollingFlag">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="displayIntegerBase">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="L0MappingTypeLabel">
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Layer 0 Mapping Type&lt;/span&gt;&lt;/p&gt;&lt;p&gt;This property is hardly used. It's only used if you're using the Toxic Landfill Tileset with an existing dust foreground.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="text">
                <string>Layer 0 Mapping Type:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_CurLayer0MappingType">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_Layer0MappingType">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="maximum">
              <number>47</number>
             </property>
             <property name="displayIntegerBase">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="L0Label">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Use existing Layer 0&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Select the foreground layer pointer.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>Use existing Layer 0:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="ComboBox_Layer0Picker">
             <property name="enabled">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_Layer0Width">
             <property name="text">
              <string>Layer 0 Width:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_Layer0Width">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="minimum">
              <number>19</number>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_Layer0Height">
             <property name="text">
              <string>Layer 0 Height:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_Layer0Height">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="minimum">
              <number>14</number>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="CheckBox_Layer0Alpha">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Layer 0 Alpha Enable&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Enable this to make color blending work.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>Enable Layer 0 Alpha Attibutes</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="ComboBox_AlphaBlendAttribute">
             <property name="enabled">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="CheckBox_Layer2Enable">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Enable Layer 2&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Enable layer X to set its mapping type from 0x00 to 0x01.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Layer X can be Layer 0 or Layer 2.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>Enable Layer 2</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="layerPriorityLabel">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Layer Priority&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;There are several available priority settings that Wario Land 4's game engine supports; choose one of them. SP+ in some of the entries means that there is a sprite priority elevation performed on all the entities. This is in addition to the priority level from their OAM data.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>Layer priority and sprite priority elevation:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="ComboBox_LayerPriority"/>
           </item>
           <item>
            <widget class="QLabel" name="roomWidthLabel">
             <property name="text">
              <string>Room Width:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="SpinBox_RoomWidth">
             <property name="minimum">
              <number>19</number>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="roomHeightLabel">
             <property name="text">
              <string>Room Height:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="SpinBox_RoomHeight">
             <property name="minimum">
              <number>14</number>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <property name="topMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_RasterType">
               <property name="text">
                <string>Raster type (Hex up to FF):</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_CurRasterType">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_RasterType">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="displayIntegerBase">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_Water">
             <property name="text">
              <string>Water (Hex up to FF):</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_Water">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="displayIntegerBase">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_BgmVolume">
             <property name="text">
              <string>BGM Volume (Hex up to FFFF):</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_BgmVolume">
             <property name="maximum">
              <number>65535</number>
             </property>
             <property name="displayIntegerBase">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_View">
       <property name="spacing">
        <number>6</number>
       </property>
       <property name="leftMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>6</number>
       </property>
       <property name="bottomMargin">
        <number>20</number>
       </property>
       <item>
        <widget class="QLabel" name="graphicViewDetailsLabel">
         <property name="text">
          <string>Tileset: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="RoomPreviewGraphicsView" name="graphicsView">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>256</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Click to see other detail(s)</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RoomPreviewGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>Dialog/RoomPreviewGraphicsView.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>RoomConfigDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>RoomConfigDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
